nistats/first_level_model.py

```diff
@@ -412,17 +431,34 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
                 design = design_matrices[run_idx]
             self.design_matrices_.append(design)
 
-            # Compute GLM
+            # Mask and prepare data for GLM
+            if self.verbose > 1:
+                t_masking = time.time()
+                print 'Starting masker computation'
             Y = self.masker_.transform(run_img)
+            if self.verbose > 1:
+                t_masking = time.time() - t_masking
+                print 'Masker took %d seconds' % t_masking
+
             if self.signal_scaling:
                 Y, _ = mean_scaling(Y, self.scaling_axis)
             if self.memory is not None:
-                mem_glm = self.memory.cache(run_glm)
+                arg_ignore = ['n_jobs']
+                mem_glm = self.memory.cache(run_glm, ignore=arg_ignore)
```

Comment:
why not? its possible recomputation would take place if the number of jobs is changed, or I misunderstand the meaning of the ignore parameter.